### PR TITLE
base code

### PR DESCRIPTION
**Pull request by tanimutomo**
main tasks
- [x] Model
- [x] Dataset
- [x] Loss function
- [x] Augmentation
- [x] Hyper params
- [x] weight init

other tasks
- [x] compute loss
- [x] attack method
- [x] make report loss more easy

Add options
- [x] scale_each
- [x] momentum
- [x] rand_target
- [x] scale_eps

### CODE REVIEW

**Comment by tanimutomo** [comment]
- weight initはなし
- datasetとtransformはそのままコードを使用
- modelもそのまま使用
- Loss funcはLPのものをそのまま使用

**Comment by tanimutomo** [comment]
attackはadvex-uarのものを利用

**Comment by tanimutomo** [comment]
default optimizer is SGD
scheduler is not used

**Comment by tanimutomo** [comment]
The command for reproducing alp accuracy of the paper
`pipenv run python src/train.py --dataset cifar10 --eps_iter 2.0 --rand_target False --scale_eps False --ct 0.5 --at 0.5 --alp 0.5`
